### ISSUE TITLE
RAM device triggers segfault in BGFX

### ISSUE THREAD

**Issue by JoakimLarsson** [mamedev/mame#5776]
Some RAM size values causes segfault, it is repeatable and happens every time:

````
work/mame$` gdb --arg ./eispcMAME21464 epc -window -ram 256K
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Registered pretty printers for UE4 classes
Reading symbols from ./eispcMAME21464...done.
(gdb) run
Starting program: /home/joakim/work/mame/eispcMAME21464 epc -window -ram 256K
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe674d700 (LWP 699)]
[New Thread 0x7fffd6df9700 (LWP 701)]
[New Thread 0x7fffd65f8700 (LWP 702)]

Thread 1 "eispcMAME21464" received signal SIGSEGV, Segmentation fault.
0x0000000000000020 in ?? ()
(gdb) bt
#0  0x0000000000000020 in ?? ()
#1  0x0000000000526365 in chain_manager::update_screen_textures(unsigned int, render_primitive*, osd_window&) ()
#2  0x00000000004bb864 in renderer_bgfx::draw(int) ()
#3  0x0000000000494683 in sdl_window_info::update() ()
#4  0x0000000000491c1b in sdl_osd_interface::update(bool) ()
#5  0x0000000001668e29 in video_manager::frame_update(bool) ()
#6  0x000000000162c705 in screen_device::vblank_begin() ()
#7  0x000000000162cf28 in screen_device::reset_origin(int, int) ()
#8  0x0000000000815220 in mc6845_device::handle_line_timer() ()
#9  0x000000000081540d in mc6845_device::device_timer(emu_timer&, unsigned int, int, void*) ()
#10 0x0000000001628370 in device_scheduler::timeslice() ()
#11 0x00000000015cf428 in running_machine::run(bool) ()
#12 0x000000000058af34 in mame_machine_manager::execute() ()
#13 0x0000000000619f67 in cli_frontend::start_execution(mame_machine_manager*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) ()
#14 0x000000000061a19e in cli_frontend::execute(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&) ()
#15 0x0000000000588e01 in emulator_info::start_frontend(emu_options&, osd_interface&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&) ()
#16 0x000000000043cc8f in main ()
(gdb) quit
```

**Comment by MooglyGuy** [comment]
I'll look into it, thanks for the report.

**Comment by JoakimLarsson** [comment]
It is a singledriver compile, in case that matters

**Comment by algestam** [comment]
I see a similar issue since today on Ubuntu 18.04. The segfault can be triggered by:
 * Starting a game with the bgfx video driver (i.e. `./mame64 -window -video bgfx outrun`)
 * Enter the in-game Slider Controls menu
 * Browse through the "Window 0, Screen 0 Effect" all to the left

```
~/development/mame$ gdb --args ./mame64 -window -video bgfx outrun
GNU gdb (Ubuntu 8.1-0ubuntu3.1) 8.1.0.20180409-git
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./mame64...done.
(gdb) run
Starting program: /home/hal/development/mame/mame64 -window -video bgfx outrun
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe4e6a700 (LWP 62442)]
[New Thread 0x7fffcd3f4700 (LWP 62443)]
[New Thread 0x7fffcd3b3700 (LWP 62444)]

Thread 1 "mame64" received signal SIGSEGV, Segmentation fault.
0x000055555d6948d7 in chain_manager::update_screen_textures(unsigned int, render_primitive*, osd_window&) ()
(gdb) bt
#0  0x000055555d6948d7 in chain_manager::update_screen_textures(unsigned int, render_primitive*, osd_window&) ()
#1  0x000055555d622882 in renderer_bgfx::draw(int) ()
#2  0x000055555d5f8fa1 in sdl_window_info::update() ()
#3  0x000055555d5f66b3 in sdl_osd_interface::update(bool) ()
#4  0x0000555561389e15 in video_manager::frame_update(bool) ()
#5  0x00005555613478b5 in screen_device::vblank_begin() ()
#6  0x000055556134b28d in screen_device::device_timer(emu_timer&, unsigned int, int, void*) ()
#7  0x0000555561343280 in device_scheduler::timeslice() ()
#8  0x00005555612e4278 in running_machine::run(bool) ()
#9  0x000055555d6f93dd in mame_machine_manager::execute() ()
#10 0x000055555d79008b in cli_frontend::start_execution(mame_machine_manager*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) ()
#11 0x000055555d79031e in cli_frontend::execute(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&) ()
#12 0x000055555d6f7035 in emulator_info::start_frontend(emu_options&, osd_interface&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&) ()
#13 0x0000555558fb8834 in main ()
(gdb) quit
```

**Comment by MooglyGuy** [comment]
The issue reported by @algestam is unrelated to the issue you've reported, @JoakimLarsson. I have fixed the issue that he reported, but I am unable to reproduce the issue that you're reporting.

On the contrary, running the epc driver with the specified options and the BGFX backend causes the assert at line 461 of schedule.cpp to trip: assert(delta == (target - exec->m_localtime).as_attoseconds());

So, something else must be going wrong here.

**Comment by algestam** [comment]
Thanks @MooglyGuy! Your fix resolves the issue I reported. Sorry for reporting it here, I thought the issues were related.

**Comment by JoakimLarsson** [comment]
Guess you are at windows @MooglyGuy? I am doing a full build now of my head which is commit  6c6b0f86a8ed5e40ead04e824dc8dbde13d1a846 , just a plain 

`make -j12` 

to see if there's any difference. I built the singel driver with   

`make SUBTARGET=eispcMAME214 SOURCES=src/mame/drivers/eispc.cpp -j12`

I think the assert comes from the driver abusing the scheduler to resolve a tight power up sequence between the keyboard and the main board. @cuavas already pointed that out for me but without it I get Keyboard error, so I need to figure out how to deal with that.

**Comment by JoakimLarsson** [comment]
A fault is definitely still there. @MooglyGuy, I can confirm the assert when setting  DEBUG=1 SYMBOLS=1 but if you make a plain compile you'll get the error as I reported. I have fetched and merged 3d4bd27551010b1d33a02e5e7473f07f58e4b03c and it doesn't fix it. I have a feeling your changes revealed a bug somewhere else, I keep looking but will not be able to properly fix this before freeze I think, so I might just disable the crashing memory configurations
